hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/webapp/NodeIDsInfo.java

```diff
@@ -58,11 +60,39 @@ public NodeIDsInfo(List<String> nodeIdsList, Resource resource) {
     this.partitionInfo = new PartitionInfo(new ResourceInfo(resource));
   }
 
+  public NodeIDsInfo(Set<String> nodeIdsList, PartitionInfo partitionInfo) {
+    this.nodeIDsList.addAll(new ArrayList<>(nodeIdsList));
```

Comment:
Do we need to create an array? Isn't addAll based on Collection?

Reply:
Thank you for your suggestion, Collection can be used, I will modify the code.